sdk/python/kfp/dsl/_container_op.py

```diff
@@ -16,11 +16,125 @@
 from . import _pipeline
 from . import _pipeline_param
 import re
-from typing import Dict
-
+from typing import Dict, List
+from abc import ABCMeta, abstractmethod
+from . import _types
+
+class BaseMeta(object):
+  __metaclass__ = ABCMeta
+  def __init__(self):
+    pass
+
+  @abstractmethod
+  def to_dict(self):
+    pass
+
+  def serialize(self):
```

Comment:
Are you sure it's going to be useful, since the metadata will be a part of larger YAML document (Argo+Metadata)